api/v1beta1/watcher_types.go

```diff
@@ -27,6 +27,11 @@ type WatcherSpec struct {
 	// Important: Run "make" to regenerate code after modifying this file
 
 	WatcherTemplate `json:",inline"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=memcached
+	// MemcachedInstance is the name of the Memcached CR that all watcher service will use.
+	MemcachedInstance string `json:"memcachedInstance"`
```

Comment:
Should it go to WatcherCommon ? I guess we will need it in all controllers to build the configuration.

Reply:
I implemented it initially following the same pattern we've done with the `Secret` field. Making it optional on Watcher but required in sub CRs, so it would be passed by Watcher when eventually it creates WatcherAPI. But thinking about it now it's different than the secret, because this won't need any processing, maybe it makes sense to move it to WatcherCommon and have it optional with a default

Reply:
updated the PR moving `MemcachedInstance` to `WatcherCommon`